src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -257,9 +257,15 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
         /// Reusing <see cref="AssemblyMetadata"/> object allows for sharing data accross these references.
         /// </para>
         /// </remarks>
+        [Obsolete("The API is obsolete, use CreateFromFile(assembly.Location) instead", error: true)]
         public static MetadataReference CreateFromAssembly(Assembly assembly)
         {
-            return CreateFromAssembly(assembly, default(MetadataReferenceProperties));
+            throw new NotImplementedException();
+        }
```

Comment:
Any code that is re-compiled between now and when we work on CoreCLR will see the Obsolete and can be updated.  Apps that have never been re-compiled, will continue to work where they currently work, and will fail on the places where we can't support them.